onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -703,42 +705,42 @@ TreeEnsembleCommon<InputType, ThresholdType, OutputType>::ProcessTreeNodeLeave(
       threshold = root->value_or_unique_weight;
       switch (root->mode()) {
         case NODE_MODE::BRANCH_LEQ:
-          root += val <= threshold || (root->is_missing_track_true() && _isnan_(val))
-                      ? root->truenode_inc_or_first_weight
-                      : root->falsenode_inc_or_n_weights;
+          root = val <= threshold || (root->is_missing_track_true() && _isnan_(val))
+                     ? root->truenode_inc_or_first_weight.ptr
+                     : root->falsenode_inc_or_n_weights.ptr;
           break;
         case NODE_MODE::BRANCH_LT:
-          root += val < threshold || (root->is_missing_track_true() && _isnan_(val))
-                      ? root->truenode_inc_or_first_weight
-                      : root->falsenode_inc_or_n_weights;
+          root = val < threshold || (root->is_missing_track_true() && _isnan_(val))
+                     ? root->truenode_inc_or_first_weight.ptr
+                     : root->falsenode_inc_or_n_weights.ptr;
           break;
         case NODE_MODE::BRANCH_GTE:
-          root += val >= threshold || (root->is_missing_track_true() && _isnan_(val))
-                      ? root->truenode_inc_or_first_weight
-                      : root->falsenode_inc_or_n_weights;
+          root = val >= threshold || (root->is_missing_track_true() && _isnan_(val))
+                     ? root->truenode_inc_or_first_weight.ptr
+                     : root->falsenode_inc_or_n_weights.ptr;
           break;
         case NODE_MODE::BRANCH_GT:
-          root += val > threshold || (root->is_missing_track_true() && _isnan_(val))
-                      ? root->truenode_inc_or_first_weight
-                      : root->falsenode_inc_or_n_weights;
+          root = val > threshold || (root->is_missing_track_true() && _isnan_(val))
+                     ? root->truenode_inc_or_first_weight.ptr
+                     : root->falsenode_inc_or_n_weights.ptr;
           break;
         case NODE_MODE::BRANCH_EQ:
-          root += val == threshold || (root->is_missing_track_true() && _isnan_(val))
-                      ? root->truenode_inc_or_first_weight
-                      : root->falsenode_inc_or_n_weights;
+          root = val == threshold || (root->is_missing_track_true() && _isnan_(val))
+                     ? root->truenode_inc_or_first_weight.ptr
+                     : root->falsenode_inc_or_n_weights.ptr;
           break;
         case NODE_MODE::BRANCH_NEQ:
-          root += val != threshold || (root->is_missing_track_true() && _isnan_(val))
-                      ? root->truenode_inc_or_first_weight
-                      : root->falsenode_inc_or_n_weights;
+          root = val != threshold || (root->is_missing_track_true() && _isnan_(val))
+                     ? root->truenode_inc_or_first_weight.ptr
+                     : root->falsenode_inc_or_n_weights.ptr;
           break;
         case NODE_MODE::LEAF:
-          break;
+          return root;
```

Comment:
Is it needed?

Reply:
It is, since this is how we exit the loop. See https://github.com/microsoft/onnxruntime/pull/17449/files#diff-098f04f4ce424d09171371b1dc5893d29e38c42a0f880639775de9e0fec67241R708.

Reply:
`break` exists the loop as well and `return root` is then found just after.

Reply:
I missed `while(1)`.